src/util/Decode.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @module Decode
+ */
+
+import { Codec, AnyJson } from '@polkadot/types/types'
+
+/**
+ * Dummy comment needed for correct doc display, do not remove.
+ */
+
+/**
+ * Checks for a non-null byte in the Uint8Array representation
+ * of the codec. This is a pretty sound indicator that the
+ * codec does not hold an empty default value. The reverse inference
+ * does not apply! Valid values could be encoded only with 0 bytes.
+ *
+ * @param codec Codec type to check for non-empty bytes.
+ * @returns True if codec is non-empty; false if it may be empty.
+ */
+export function hasNonNullByte(codec: Codec): boolean {
+  return !codec.toU8a().some(e => e !== 0)
+}
+
+export function decodeArray(codec: Codec[]): AnyJson[] {
+  return codec.map(el => {
+    if (el instanceof Array) {
+      return decodeArray(el)
+    }
+    if (el.isEmpty && el.toJSON) {
```

Comment:
Some Codec types return true for `isEmpty` on falsy values; a U64(0) would return true for `isEmpty`, as would a Vec(..., []). If a decoded value is falsy, I see no point in replacing it with `null`. In fact, this may even lead to buggy behavior, I thought. 
Do you think that's to fiddly ? I should deffo explain it well in a doc comment

Reply:
Should definitely be explained. But I still think the check of `el.toJSON` is insufficient here in that it should be at least `el.toJSON()`?!

Reply:
f*** it, decided to remove the function again and reverted `asArray()` - there's more problems to it, and as long as I can't produce a clean implementation, I'd rather not add this at all. It's not part of the ticket anyways.